lambdas/services/lloyd_george_validator.py

```diff
@@ -1,7 +1,44 @@
+import re
+
+from lambdas.models.nhs_document_reference import NHSDocumentReference
+
+
 class LGFileTypeException(ValueError):
-    """One or more of the files do not match the required file type."""
+    """One or more of the files do not match the required file type"""
     pass
 
-def validate_lg_file_type(file_type):
+class LGFileNameException(ValueError):
+    """One or more of the files do not match naming convention"""
+    pass
+
+class LGInvalidFilesException(Exception):
+    """One or more of the files are not valid"""
+    pass
+
+def validate_lg_file_type(file_type: str):
     if file_type != 'application/pdf':
         raise LGFileTypeException
+
+def validate_file_name(name: str):
+    lg_regex = r'[0-9]+of[0-9]+_Lloyd_George_Record_\[[A-Za-z ]+]_\[[0-9]{10}]_\[\d\d-\d\d-\d\d\d\d].pdf'
```

Comment:
Will this A-Za-z include special characters such as e, è, é, ê, ë for people with names that include these?

Reply:
I'm not sure if we want to include those. I'll check

Reply:
How about people with hyphenated names like Catherine Zeta-Jones or Sophie Ellis-Bextor, this will not match those:

1of2_Lloyd_George_Record_[Sophie Ellis-Bextor]_[1234567890]_[25-12-2019].pdf
1of2_Lloyd_George_Record_[Harvey Guillén]_[1234567890]_[25-12-2019].pdf

This SO looks promising:

https://stackoverflow.com/questions/24676691/whats-a-good-regex-to-include-accented-characters-in-a-simple-way

so I did one myself, this one matches both edge cases:

[0-9]+of[0-9]+_Lloyd_George_Record_\[[A-Za-z À-ÿ\-]+]_\[[0-9]{10}]_\[\d\d-\d\d-\d\d\d\d]

Reply:
Would be nice to see tests for those cases too :)

Reply:
`[0-9]+of[0-9]+_Lloyd_George_Record_\[[A-Za-z À-ÿ\-]+]_\[[0-9]{10}]_\[\d\d-\d\d-\d\d\d\d]`

^ since the above example was autoformatted by github to _italicise_ things in between 2 _'s

Reply:
After the refinement session today, I suggest we do something like this to extract all wanted info first:

```
def extract_info_from_filename(filename: str) -> dict:
    page_number = r"(?P<page_no>[1-9][0-9]*)"
    total_page_number = r"(?P<total_page_no>[1-9][0-9]*)"
    patient_name = r"(?P<patient_name>[A-Za-z À-ÿ\-]+)"
    nhs_number = r"(?P<nhs_number>\d{10})"
    date_of_birth = r"(?P<dob>\d\d-\d\d-\d\d\d\d)"
    
    lg_regex = rf"{page_number}of{total_page_number}_Lloyd_George_Record_\[{patient_name}\]_\[{nhs_number}\]_\[{date_of_birth}\].pdf"

    if match := re.fullmatch(lg_regex, filename):
        return match.groupdict()
    else:
        raise LGInvalidFilesException('One or more of the files do not match naming convention')
```

Motive of having this:
1. Relatively easy to change the regex of one part in the filename e.g. patient name, date of birth
2. The logic of checking page numbers can be separately from the logic of regex matching
3. Relatively easy to add more further validation e.g. check every file got the same patient name, check date of birth is a valid date etc

Reply:
Done